### PR TITLE
[draft] fix(require-cycles): remove require cycle between Appbar and AppbarHeader

### PR DESCRIPTION
**Pull request by FrederickEngelhardt**
### Summary

Fixes require cycle Appbar -> AppbarHeader -> Appbar.

FYI there are a few more remaining items:
`npx dpdm ./src/index.tsx --tree false`

Outputs this:
```
  1) src/components/Appbar/AppbarContent.tsx -> src/components/Appbar/utils.ts
  2) src/components/TextInput/TextInput.tsx -> src/components/TextInput/Adornment/TextInputAffix.tsx -> src/components/TextInput/helpers.tsx -> src/components/TextInput/types.tsx
```

This issue exists on release [5.3.1](https://github.com/callstack/react-native-paper/tree/v5.3.1)

### Test plan
1. Did not test this. This is a draft PR. Hoping to have some feedback.

### CODE REVIEW

File: src/components/Appbar/Appbar.props.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    top?: number;
+    left?: number;
+    right?: number;
+  };
+  /**
+   * @optional
+   */
+  theme?: ThemeProp;
+  style?: Animated.WithAnimatedValue<StyleProp<ViewStyle>>;
+};
```
Comment (DimitarNestorov) [inline src/components/Appbar/Appbar.props.ts:47]: Thank you for the PR @FrederickEngelhardt!

Due to the way the docs are currently generated the props need to be in the same file: https://output.circle-artifacts.com/output/job/b45a975a-fbbd-4697-bbc8-293acf79106b/artifacts/0/docs/docs/components/Appbar/

Comment (FrederickEngelhardt) [reply src/components/Appbar/Appbar.props.ts:47]: @DimitarNestorov not following, the comment link seems to redirect to this PR only.

Comment (lukewalczak) [reply src/components/Appbar/Appbar.props.ts:47]: The comment above from `callstack-bot` is containing the link with the documentation preview, where you can observe that `Props` section for the `Appbar` _(for which you extracted props to the separated file)_ are basically empty.

**Comment by lukewalczak** [comment]
Closing in favor of: https://github.com/callstack/react-native-paper/pull/4056

**Pull request closed without merging**